infrastructure/restapi/src/main/java/tech/pegasys/teku/infrastructure/restapi/endpoints/JavalinRestApiRequest.java

```diff
@@ -47,6 +47,10 @@ public class JavalinRestApiRequest implements RestApiRequest {
   @Override
   @SuppressWarnings({"TypeParameterUnusedInFormals"})
   public <T> T getRequestBody() throws JsonProcessingException {
+    if (!metadata.isRequiredRequestBody()) {
+      throw new RuntimeException(
+          "Optional request body configured, use getOptionalRequestBody() instead");
+    }
```

Comment:
this will end up a 500 response...

Reply:
Now that I thought about it, I don't know if enforcing this here makes sense. This error will happen when the application is running and the user has nothing to do with what method is being used here (the developer does).

I believe the right thing to do to enforce the usage of required/optional would be a compile-time warning. This way the developer can ensure they use the right method.

Reply:
My point here was to catch something in the first ever test you write if you are using getRequestBody without optional

Reply:
Fair point! We just need to make sure this error message does not make its way to the user when calling the API 👍

Reply:
So my intention was that only developer will see it. It could go outside if someone makes 0 tests on new endpoint and we merge it

Reply:
Yeah, that's what I ended up settling on... If we have a new RPC with zero tests in prod we will probably have bigger problems than the error message to worry about... :P